scheduler/servicestatemanager/servicestatemanager.go

```diff
@@ -66,21 +66,21 @@ type ServiceStateManager interface {
 // ServiceStateChangeBatch represents a batch of services with the same
 // desired state that will be operated on by a ServiceStateManager
 type ServiceStateChangeBatch struct {
-	Services     map[string]CancellableService
+	Services     map[string]*CancellableService
 	DesiredState service.DesiredState
 	Emergency    bool
 }
 
 // CancellableService is a service whose scheduling may be canceled by a channel
 type CancellableService struct {
-	sync.Mutex
 	*service.Service
-	cancel chan interface{}
-	C      <-chan interface{}
+	cancel     chan interface{}
+	C          <-chan interface{}
+	cancelLock sync.Mutex
```

Comment:
Need to make this a pointer, otherwise it will be a pass by value and be almost useless.